presto-main/src/main/java/io/prestosql/operator/scalar/timestamp/VarcharToTimestampCast.java

```diff
@@ -71,47 +57,17 @@ public static long castToShort(@LiteralParameter("p") long precision, ConnectorS
 
     @LiteralParameters({"x", "p"})
     @SqlType("timestamp(p)")
-    public static LongTimestamp castToLong(@LiteralParameter("p") long precision, ConnectorSession session, @SqlType("varchar(x)") Slice value)
+    public static LongTimestamp castToLong(@LiteralParameter("p") long precision, @SqlType("varchar(x)") Slice value)
     {
-        // This accepts value with or without time zone
-        if (session.isLegacyTimestamp()) {
-            try {
-                return castToLegacyLongTimestamp((int) precision, session.getTimeZoneKey(), trim(value).toStringUtf8());
-            }
-            catch (IllegalArgumentException e) {
-                throw new PrestoException(INVALID_CAST_ARGUMENT, "Value cannot be cast to timestamp: " + value.toStringUtf8(), e);
-            }
-        }
-
         try {
-            return castToLongTimestamp((int) precision, trim(value).toStringUtf8(), timezone -> ZoneOffset.UTC);
+            return castToLongTimestamp((int) precision, trim(value).toStringUtf8());
         }
         catch (IllegalArgumentException e) {
             throw new PrestoException(INVALID_CAST_ARGUMENT, "Value cannot be cast to timestamp: " + value.toStringUtf8(), e);
         }
     }
 
-    public static long castToLegacyShortTimestamp(int precision, TimeZoneKey timeZoneKey, String value)
-    {
-        return castToShortTimestamp(precision, value, timezone -> {
-            if (timezone == null) {
-                return timeZoneKey.getZoneId();
-            }
-            return ZoneId.of(timezone);
-        });
-    }
-
-    private static LongTimestamp castToLegacyLongTimestamp(int precision, TimeZoneKey timeZoneKey, String value)
-    {
-        return castToLongTimestamp(precision, value, timezone -> {
-            if (timezone == null) {
-                return timeZoneKey.getZoneId();
-            }
-            return ZoneId.of(timezone);
-        });
-    }
-
-    private static long castToShortTimestamp(int precision, String value, Function<String, ZoneId> zoneId)
+    public static long castToShortTimestamp(int precision, String value)
```

Comment:
Why public? This is supposed to be an implementation detail of this class.

Reply:
They replace the old legacy method used in tests.  I will mark both as visible for testing.